### PR TITLE
Update source index package version

### PR DESCRIPTION
**Pull request by eerhardt**
The current official builds are failing with

"Unsupported log file format. Latest supported version is 17, the log file has version 18."

Updating to the latest version.

### CODE REVIEW

File: eng/common/templates/job/source-index-stage1.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 parameters:
   runAsPublic: false
-  sourceIndexPackageVersion: 1.0.1-20231213.4
+  sourceIndexPackageVersion: 1.0.1-20240129.1
```
Comment (wtgodbe) [inline eng/common/templates/job/source-index-stage1.yml:3]: @eerhardt this change should also be made in arcade - this file is in `eng/common`, so any changes here will be overwritten by the next arcade update

**Review by wtgodbe — APPROVED** [review]

**Comment by wtgodbe** [comment]
Additionally, we'll probably have to make a change in the source-indexer repo to ingest a new version of this package - it hasn't been updated recently enough to make me think the latest source-indexer package will work: https://github.com/dotnet/source-indexer/blob/b44bfa8bd7733889cd352316f325ff3ef114328d/src/SourceBrowser/src/Directory.Packages.props#L45

**Comment by wtgodbe** [comment]
https://github.com/dotnet/source-indexer/pull/133

File: eng/common/templates/job/source-index-stage1.yml (hunk 1 above)
Comment (eerhardt) [reply eng/common/templates/job/source-index-stage1.yml:3]: Good call. Will send an arcade PR.

**Comment by wtgodbe** [comment]
https://dev.azure.com/dnceng/internal/_build/results?buildId=2365147&view=results will produce the new package

File: eng/common/templates/job/source-index-stage1.yml (hunk 1 above)
Comment (wtgodbe) [inline eng/common/templates/job/source-index-stage1.yml:3]:
```suggestion
  sourceIndexPackageVersion: 1.0.1-20240129.2
```

**Pull request merged**